runtime/src/bank.rs

```diff
@@ -3864,6 +3865,22 @@ mod tests {
         let bank1 = Bank::new_from_parent(&bank0, &Pubkey::default(), 1);
         bank1.freeze();
         let bank1_hash = bank1.hash();
+        // Checkpointing should always result in a new state
         assert_ne!(bank0_hash, bank1_hash);
     }
+
+    #[test]
+    fn test_ticks_change_state() {
+        let (genesis_config, _) = create_genesis_config(500);
+        let bank = Arc::new(Bank::new(&genesis_config));
+        let bank1 = new_from_parent(&bank);
+        let hash1 = bank1.hash_internal_state();
+        // ticks don't change its state unless a block boundary is crossed
+        bank1.register_tick(&Hash::default());
+        assert_eq!(bank1.hash_internal_state(), hash1);
+        for _ in 0..genesis_config.ticks_per_slot {
```

Comment:
nits: Even better, put the above line `assert_eq!(bank1.hash_internal_state(), hash1);` here inside the `for` loop so that we can assert that the hash doesn't change between 0th and `(genesis_config.ticks_per_slot-1)`-th ticks.

Reply:
smart. will do.

Reply:
Err, we are doing `register_tick` before, so there is need to remove it or adjust the loop condition. I prefer just the former. :)

Reply:
@sagar-solana ^^ I hope you also noticed this already. :)

Reply:
yup, i removed it :)